src/app/games/new-game/tests/NewGameDummyMinimax.spec.ts

```diff
@@ -1,36 +1,64 @@
 import { RulesUtils } from 'src/app/jscaip/tests/RulesUtils.spec';
-import { NewGameDummyMinimax } from '../NewGameDummyMinimax';
+import { NewGameDummyMinimax, NewGameHeuristic, NewGameMoveGenerator } from '../NewGameDummyMinimax';
 import { NewGameMove } from '../NewGameMove';
-import { NewGameNode, NewGameRules } from '../NewGameRules';
+import { NewGameNode } from '../NewGameRules';
 import { NewGameState } from '../NewGameState';
 
 /**
- * These are the tests for the minimax.
- * We want to test two things: the list of possible moves and the scoring function.
+ * These are the tests for the move generator.
+ * We want to test that it gives us the expected moves.
+ * Typically, this can be done by checking the number of moves available on the first turn of a game.
  */
-describe('NewGameDummyMinimax', () => {
+describe('NewGameMoveGenerator', () => {
 
-    let minimax: NewGameDummyMinimax;
+    let moveGenerator: NewGameMoveGenerator;
 
     beforeEach(() => {
-        minimax = new NewGameDummyMinimax(NewGameRules.get(), 'NewGameDummyMinimax');
+        moveGenerator = new NewGameMoveGenerator();
     });
     it('should have all move options', () => {
         // Given an initial node
         const initialState: NewGameState = NewGameState.getInitialState();
         const node: NewGameNode = new NewGameNode(initialState);
 
         // When computing the list of moves
-        const moves: NewGameMove[] = minimax.getListMoves(node);
+        const moves: NewGameMove[] = moveGenerator.getListMoves(node);
 
         // Then there should be this many moves
         expect(moves.length).toBe(0);
     });
+});
+
+
+/**
+ * These are the tests for the heuristic.
+ * We want to test that it gives some value on some boards, or rather that it assigns higher values to a board
+ * compared to another one.
+ * We can rely on RulesUtils' functions to achieve this.
+ */
+describe('NewGameHeuristic', () => {
+
+    let heuristic: NewGameHeuristic;
+
+    beforeEach(() => {
+        heuristic = new NewGameHeuristic();
+    });
     it('should have some score', () => {
         /**
          * To test scores, most of the time you want to rely on `RulesUtils.expectSecondStateToBeBetterThanFirstFor`.
          */
         const initialState: NewGameState = NewGameState.getInitialState();
-        RulesUtils.expectStatesToBeOfEqualValue(minimax, initialState, initialState);
+        RulesUtils.expectStatesToBeOfEqualValue(heuristic, initialState, initialState);
+    });
+});
+
+
+/**
+ * Because we tested the move generator and the heuristic, the minimax should be correct by default.
+ * Hence, we only have to test that it is created correctly.
+ */
+describe('NewGameDummyMinimax', () => {
```

Comment:
if you want to test that the minimax at level N indeed choose move.x as a best move, it's here ! (and provide sample)

Reply:
Oui, souhaitables !
Je recommande de tester le tout ! La redondance des tests n'est PAS un soucis je crois !

Reply:
L'explicite vaut mieux que faire le mathématicien !

Reply:
agreed, ajouté du code et commentaires pour ça donc